lib/std/smartptrs.nim

```diff
@@ -0,0 +1,130 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2021 Nim contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+
+## C++11 like smart pointers. They always use the shared allocator.
+import std/isolation
+
```

Comment:
needs a few runnableExamples illustrating how to use each smart pointer (maybe 1 runnnableExamples per smart pointer type)

Reply:
Will do